src/main/kotlin/fi/aalto/cs/apluscourses/generator/APlusModuleBuilder.kt

```diff
@@ -0,0 +1,272 @@
+                }
+            }
+            courseList.putClientProperty(AnimatedIcon.ANIMATION_IN_RENDERER_ALLOWED, true)
+            courseList.selectionMode = ListSelectionModel.SINGLE_SELECTION
+
+            courseList.addListSelectionListener(ListSelectionListener { e ->
+                val url = courseList.selectedValue?.url ?: return@ListSelectionListener
+                courseConfigUrl.set(url)
+            })
+
+            application.service<CoursesFetcher>()
+                .fetchCourses { courses -> courseList.setListData(courses.toTypedArray()) }
+        }
+
+        override fun getComponent(): JComponent {
+            return panel {
+                panel {
+                    row {
+                        text("A project linked to an A+ LMS course").applyToComponent {
+                            foreground = JBUI.CurrentTheme.ContextHelp.FOREGROUND
+                        }
+                    }
+                    row {
+                        text("Select a course:")
+                    }
+                    row {
+                        cell(courseList).resizableColumn().align(AlignX.FILL)
+                    }
+                    row("Configuration file URL:") {
+                        textField()
+                            .bindText(courseConfigUrl)
+                            .resizableColumn()
+                            .align(AlignX.FILL)
+                    }
+                    row {
+                        text("").bindText(errorMessage).applyToComponent {
+                            foreground = JBUI.CurrentTheme.NotificationError.borderColor()
+                        }
+                    }
+                }.customize(UnscaledGaps(20, 20, 20, 20))
+            }
+        }
+
+        override fun updateDataModel() {}
+
+        override fun validate(): Boolean {
+            val url = courseConfigUrl.get()
+            if (url.isEmpty()) {
+                errorMessage.set("Please select a course.")
+                return false
+            }
+            val courseConfig = application.service<CoursesFetcher>().fetchCourse(url)
+            if (courseConfig == null) {
+                errorMessage.set("Invalid URL.")
+                return false
+            }
+            errorMessage.set("")
+            this@APlusModuleBuilder.courseConfig = courseConfig
+            this@APlusModuleBuilder.courseConfigUrl = url
+            this@APlusModuleBuilder.programmingLanguage = courseList.selectedValue.language ?: ""
+            return true
+        }
+    }
+
+    inner class CourseSettingsStep(
+        val wizardContext: WizardContext
+    ) : ModuleWizardStep() {
+        private var mainPanel = panel {}
+        private val selectedLanguage = AtomicProperty<String>("")
+        private val dontImportSettings = AtomicProperty(false)
+        private var selectedSdk: AtomicProperty<ProjectWizardJdkIntent>? = null
```

Comment:
Should this be:
```kotlin
private var selectedSdk = AtomicProperty<ProjectWizardJdkIntent?>(null)
```

Reply:
Unfortunately bindItem expects an AtomicProperty<ProjectWizardJdkIntent> not AtomicProperty<ProjectWizardJdkIntent?>